docs/examples/tunnel-with-reverse-proxy/README.md

```diff
@@ -0,0 +1,162 @@
+      - `<account-id>`: the account ID associated with the cloudflare zone you are managing.
+   - in `manifests/hello/ingress.yaml`
+     - `<domain>`: the domain associated with the cloudflare zone you are managing.
+4. deploy the cluster tunnel and tunnelbinding
+```shell
+kubectl apply -f manifests/cloudflare-operator/
+```
+5. deploy the reverse proxy
+```shell
+kustomize build --enable-helm manifests/ingress-nginx | kubectl apply -f - 
+```
+6. deploy the example application
+```shell
+kubectl apply -f manifests/hello/
+```
+7. You should now be able to access the service on `hello.<domain>`
+
+## Extending this example
+
+Now that we are using an arbitrary reverse proxy for routing traffic into the cluster, we can configure it as if we were using a cloud load balancer.
+This means you can use whatever constructs are available to your reverse proxy implementation.
+
+### Custom SSO
+
+One can configure SSO with a custom provider.
+This example uses ingress-nginx and authelia
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: <app>
+  annotations:
+    cert-manager.io/cluster-issuer: "letsencrypt-prod"
+    # in order to direct cloudflare tunnel to ingress-nginx, we have a single tunnelBinding resource in ingress-nginx.
+    # this is what controls ingress to ingress-nginx and allows us to put 2fa infront of apps that don't support it natively.
+    nginx.ingress.kubernetes.io/rewrite-target: /
+    nginx.ingress.kubernetes.io/auth-method: "GET"
+    nginx.ingress.kubernetes.io/auth-url: "http://authelia.authelia.svc.cluster.local:8080/api/authz/auth-request"
+    nginx.ingress.kubernetes.io/auth-signin: "https://authelia.<domain>?rm=$request_method"
+    nginx.ingress.kubernetes.io/auth-response-headers: "Remote-User,Remote-Name,Remote-Groups,Remote-Email"
+--- SNIP ---
+```
+
+### Using cloudflared for routing as well as ingress-nginx
+
+Using this approach does not prohibit additional routes.
+This is useful if you are migrating to/from one approach, or have different requirements for different endpoints.
+For example, you may want some endpoints to bypass your reverse proxy for some reason.
+
+```yaml
+apiVersion: networking.cfargotunnel.com/v1alpha1
+kind: TunnelBinding
+metadata:
+  name: authelia
+  namespace: authelia
+subjects:
+  - name: authelia # the fqdn authelia.<domain> maps to this service
+tunnelRef:
+  kind: ClusterTunnel
+  name: example-tunnel
+---
+apiVersion: networking.cfargotunnel.com/v1alpha1
+kind: TunnelBinding
+metadata:
+  name: ingress-nginx
+subjects:
+  # At this time cloudflare-operator does not deterministically sort wildcards to the end of cloudflared's config file.
+  # This is important because cloudflared goes through the list in order, and this ingress being before the authelia ingress
+  # would mean authelia is impossible to reach.
+  # By prefixing with `zz_` we ensure this is the last route in the cloudflared config file
+  - name: zz_wildcard
```

Comment:
It is the name of the tunnelBinding that is sorted, not the subject.

Reply:
moved the comment :)